src/main/java/application/FormplayerAuthFilter.java

```diff
@@ -57,7 +59,15 @@
     protected void doFilterInternal(HttpServletRequest req, HttpServletResponse response, FilterChain filterChain)  {
         try {
             FormplayerHttpRequest request = getRequestIfAuthorized(req);
-            filterChain.doFilter(request, response);
+            ContentCachingResponseWrapper responseWrapper = new ContentCachingResponseWrapper(response);
```

Comment:
@knguyen142 Added a quick wrapper setup here which I've tested does what you'd expect.

It provides a clear copy of the concrete response body in a context where the user details should be available to be tacked onto the request, and the URL should be available for the log.

I think this is the cleanest entry point. My only concern is how this handles the edge cases around a couple of the large payload responses (Form beans in particular), since you get a pretty large response back for every change, but that's more of an architecture question for danny

Reply:
Thanks @ctsims. I'm playing around with it and do see that it gets the response. However, I'm having issues getting access to the response body in the logging aspect. It appears that when line 67 is run, the logging aspect has already completed. 

Are you suggesting we should move the logging logic from the logging aspect to the intercepting filter chain, in which case we should create another filter?

Reply:
I found https://github.com/dimagi/formplayer/blob/2cd847a09ac05afc88687d8771b95006ee760d77/src/main/java/application/Application.java#L49, would it be something like that?

Reply:
Hey, sorry for the confusion, I was in fact suggesting replacing the aspect driven approach with using the filter I sent over as the entry point for the logging. Will catch up on your changes since now.